core/internal/cltest/factories.go

```diff
@@ -477,13 +479,17 @@ func MustInsertPipelineRunWithStatus(t *testing.T, db *sqlx.DB, pipelineSpecID i
 	default:
 		t.Fatalf("unknown status: %s", status)
 	}
-	require.NoError(t, db.Get(&run, `INSERT INTO pipeline_runs (state,pipeline_spec_id,finished_at,outputs,all_errors,fatal_errors,created_at) VALUES ($1, $2, $3, $4, $5, $6, NOW()) RETURNING *`, status, pipelineSpecID, finishedAt, outputs, allErrors, fatalErrors))
+	require.NoError(t, db.Get(&run, `INSERT INTO pipeline_runs (state,pipeline_spec_id,pruning_key,finished_at,outputs,all_errors,fatal_errors,created_at) VALUES ($1, $2, $3, $4, $5, $6, $7, NOW()) RETURNING *`, status, pipelineSpecID, jobID, finishedAt, outputs, allErrors, fatalErrors))
 	return run
 }
 
 func MustInsertPipelineSpec(t *testing.T, db *sqlx.DB) (spec pipeline.Spec) {
 	err := db.Get(&spec, `INSERT INTO pipeline_specs (dot_dag_source,created_at) VALUES ('',NOW()) RETURNING *`)
 	require.NoError(t, err)
+	_, err = db.Exec(`SET CONSTRAINTS fk_job_pipeline_spec_job DEFERRED`)
```

Comment:
@vyzaldysanchez Lots of `SET CONSTRAINTS DEFERRED` everywhere -- does that mean that we just won't test this constraint anywhere? What confidence do we have that this is doing the right thing?

Reply:
The statements were added in the context that not all of this methods helpers in tests, and some tests, are not expecting to work directly creating jobs, which appear to be out of the scope. I saw the practice in some places, and in the same places I set the constraint, and I just followed it. You can for example see multiple instances of `SET CONSTRAINTS pipeline_runs_pipeline_spec_id_fkey DEFERRED` through the test code, which refers to the `pipeline_runs` table pipeline_spec_id column.

And the constraint is actually being tested, so I'm confident that it is doing the right thing.

Reply:
Should I just force the job to be created on tests? I can modify this to create an arbitrary job to comply with the constraint.

Reply:
Hmmmm 🤔 Do we need a job_pipeline_spec at all in this case? I think we should still be able to create a pipeline_spec without a job_pipeline_spec.

Reply:
I figured you were right here, so I took a look. The only place using this functionality that requires the `job_pipeline_specs` record to exists, is the pruning tests, so I extracted the whole thing from here, now we are just doing it on the pruning test instead.

Good call.

Though my question is, in which case we would like specs to be created without the relationship to jobs?